src/hooks/use-promise.ts

```diff
@@ -1,31 +1,47 @@
-import { useState, useCallback } from 'react';
+import { useMemo, useState, useEffect } from 'react';
 
-export function usePromise<TResult>(promise: () => Promise<TResult>) {
+export function usePromise<TResult, TErrorResult extends {}>(promise: (abortSignal?: AbortSignal) => Promise<TResult>) {
     const [data, setData] = useState<TResult | undefined>(undefined);
     const [loading, setLoading] = useState(false);
-    const [error, setError] = useState(false);
-    const [errorCode, setErrorCode] = useState<number | undefined>(undefined);
+    const [error, setError] = useState<{ code: number; data: TErrorResult } | undefined>(undefined);
 
-    const trigger = useCallback(async () => {
+    const resetPromiseData = () => {
```

Comment:
Snyggt med resetPromiseData! då kan vi börja använda denna hook igen eftersom vi gjort en variant av denna i cardPortal!

Reply:
Translate : "Nice with resetPromiseData! then we can start using this hook again because we made a variation of this in cardPortal!"